### PR TITLE
Extending execution context for retries

### LINKED ISSUE #373: Add support for RetryContext
Issue: Azure/azure-webjobs-sdk#2595 added support to access retry context within execution context.

This tracking item to expose the same property in python functions. Currently blocked on Azure/azure-functions-host#7048

### PR DESCRIPTION
**Pull request by alrod**
resolves #373

### CODE REVIEW

File: azure-functions-language-worker-protobuf/src/proto/FunctionRpc.proto
Comment (mhoeger) [inline azure-functions-language-worker-protobuf/src/proto/FunctionRpc.proto]: The default value for retry_count and max_retry_count will be 0 (because it is a [proto3 numeric type](https://developers.google.com/protocol-buffers/docs/proto3#default)). So even if it isn't explicitly set, the customer will see `context.executionContext.maxRetryCount` => 0 and `context.executionContext.retryCount` => 0. 

I think this is ok and desirable behavior but I just want to double check!

If not, please use [NullableDouble](https://github.com/Azure/azure-functions-language-worker-protobuf/blob/821cff191bd59ec224ce2886af5df039fedbd997/src/proto/shared/NullableTypes.proto#L14) or create and use some NullableInt

File: src/public/Interfaces.ts
Comment (mhoeger) [inline src/public/Interfaces.ts]: Could you add a bit about how to configure retry count? Or a link to docs?

File: src/public/Interfaces.ts
Comment (mhoeger) [inline src/public/Interfaces.ts]: Same for this comment

**Review by mhoeger — COMMENTED** [review]
A few comments, mostly looks good!!

File: azure-functions-language-worker-protobuf/src/proto/FunctionRpc.proto
Comment (alrod) [reply azure-functions-language-worker-protobuf/src/proto/FunctionRpc.proto]: I wrapped retry properties into `RetryContext`. Exactly the same we had in C# `ExecutionContext`.

**Review by mhoeger — APPROVED** [review]
LGTM! Thanks for the changes and for the interface comments!!

Before merging this PR though, could you please merge this one: https://github.com/Azure/azure-functions-language-worker-protobuf/pull/53#pullrequestreview-586075083
And then pull updates from azure-functions-language-worker-protobuf to this repo following [these steps](https://github.com/Azure/azure-functions-language-worker-protobuf#pulling-updates)? These steps make sure that proto files don't get out of sync between workers.

Will approve again if that change dismisses that review!

**Comment by alrod** [comment]
> LGTM! Thanks for the changes and for the interface comments!!
> 
> Before merging this PR though, could you please merge this one: [Azure/azure-functions-language-worker-protobuf#53 (review)](https://github.com/Azure/azure-functions-language-worker-protobuf/pull/53#pullrequestreview-586075083)
> And then pull updates from azure-functions-language-worker-protobuf to this repo following [these steps](https://github.com/Azure/azure-functions-language-worker-protobuf#pulling-updates)? These steps make sure that proto files don't get out of sync between workers.
> 
> Will approve again if that change dismisses that review!

1. I updated the prorofile from protobuf repository.
2. Also I added new Exception interface for retry context. (it is the same to `IRpcException`)
@mhoeger, please review once again.

**Review by mhoeger — APPROVED** [review]
LGTM!✅

**Pull request merged**
